### PR TITLE
add annotation_results parameter for exit tasks, and save annotations

### PR DESCRIPTION
**Pull request by rcerven**
in exit task results instead of pipelinerun annotations

* CLOUDBLD-10949

Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] Python type annotations added to new code
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates
- n/a New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/plugins/store_metadata.py [hunk 1]
```diff
@@ -151,17 +148,16 @@ def run(self):
...
-        try:
-            self.workflow.osbs.update_annotations_on_build(pipeline_run_name, annotations)
-            self.log.debug("annotations on pipeline run '%s' were updated", pipeline_run_name)
-            self.log.debug("annotations on pipeline run are : '%s'",
-                           self.workflow.osbs.get_build_annotations(pipeline_run_name))
-        except OsbsResponseException:
-            self.log.debug("annotations: %r", annotations)
-            raise
+        if self.workflow.annotations_result:
+            annotations_result = {'plugins-metadata': annotations['plugins-metadata']}
```
Comment (chmeliik) [inline atomic_reactor/plugins/store_metadata.py:156]: How big is plugins-metadata after converting to JSON? Do we need to worry about hitting the 4096 byte limit of Tekton results?

File: atomic_reactor/tasks/binary.py [hunk 2]
```diff
@@ -106,6 +112,24 @@ class BinaryExitTask(plugin_based.PluginBasedTask[TaskParams]):
...
+            pipeline_run_name=self._params.pipeline_run_name,
+            source=self._params.source,
+            user_params=self._params.user_params,
+            reactor_config_path=self._params.config_file,
+            # Set what plugins to run and how
+            plugins_conf=self.plugins_conf,
+            keep_plugins_running=self.keep_plugins_running,
+            annotations_result=self._params.annotations_result,
+        )
+        return workflow
```
Comment (chmeliik) [inline atomic_reactor/tasks/binary.py:131]: Let's not duplicate the logic. Use `super().prepare_workflow()` to get workflow, modify as needed, return workflow
```suggestion
        workflow = super().prepare_workflow()
        workflow.annotations_result = self._params.annotations_result
        return workflow
```

File: atomic_reactor/tasks/sources.py [hunk 3]
```diff
@@ -28,21 +28,13 @@ def source(self) -> source.Source:
...
-
-    def prepare_workflow(self) -> inner.DockerBuildWorkflow:
-        """After preparing the workflow as usual, fully initialize the root build dir.
-
-        Unlike the binary container container workflow, the platforms to be used for the build
-        are known in advance here (more accurately, source containers do not have a platform).
-        """
-        workflow = super().prepare_workflow()
-        workflow.build_dir.init_build_dirs(["noarch"], self._params.source)
-        return workflow
```
Comment (chmeliik) [inline atomic_reactor/tasks/sources.py:42]: Why was this removed?

File: atomic_reactor/tasks/sources.py [hunk 4]
```diff
@@ -65,3 +67,26 @@ class SourceExitTask(SourceBuildBaseTask):
...
+                source=self._params.source,
+                user_params=self._params.user_params,
+                reactor_config_path=self._params.config_file,
+                # Set what plugins to run and how
+                plugins_conf=self.plugins_conf,
+                keep_plugins_running=self.keep_plugins_running,
+                annotations_result=self._params.annotations_result,
+            )
+        workflow.build_dir.init_build_dirs(["noarch"], self._params.source)
+        return workflow
```
Comment (chmeliik) [inline atomic_reactor/tasks/sources.py:92]: Same as the binary build task, let's not duplicate it

File: atomic_reactor/plugins/store_metadata.py (hunk 1 above)
Comment (rcerven) [reply atomic_reactor/plugins/store_metadata.py:156]: it is working fine now with full build

File: atomic_reactor/tasks/binary.py (hunk 2 above)
Comment (rcerven) [reply atomic_reactor/tasks/binary.py]: ok

File: atomic_reactor/tasks/sources.py (hunk 3 above)
Comment (rcerven) [reply atomic_reactor/tasks/sources.py:42]: it's back

File: atomic_reactor/plugins/store_metadata.py (hunk 1 above)
Comment (chmeliik) [reply atomic_reactor/plugins/store_metadata.py:156]: And what is the size? Are we barely getting by, in danger of breaking any time? Or do we have lots of extra space?

Comment (rcerven) [reply atomic_reactor/plugins/store_metadata.py:156]: I'll have to check

File: atomic_reactor/tasks/sources.py (hunk 3 above)
Comment (chmeliik) [reply atomic_reactor/tasks/sources.py:42]: Could you also put it back in the same place, so that the diff will show what changed?

File: atomic_reactor/plugins/store_metadata.py (hunk 1 above)
Comment (rcerven) [reply atomic_reactor/plugins/store_metadata.py:156]: currently with all plugins completed it is 3400

**Review by mkosiarc — APPROVED** [review]

**Review by lkolacek — APPROVED** [review]

**Comment by rcerven** [comment]
rn updated
